common-npm-packages/codeanalysis-common/Common/SpotbugsTool.ts

```diff
@@ -0,0 +1,144 @@
+import { BuildOutput } from './BuildOutput';
+import { ModuleOutput } from './ModuleOutput';
+import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+import { BaseTool } from './BaseTool';
+
+import path = require('path');
+import fs = require('fs');
+import xml2js = require('xml2js');
+
+import * as tl from 'azure-pipelines-task-lib/task';
+
+const defaultPluginVersion: string = '4.7.0';
+
+/**
+ * An object that is able to configure the build to run SpotBugs and identify and parse SpotBugs reports
+ *
+ * @export
+ * @class PmdReportParser
+ * @implements {IAnalysisToolReportParser}
+ */
+export class SpotbugsTool extends BaseTool {
+    constructor(buildOutput: BuildOutput, boolInputName: string) {
+        super('SpotBugs', buildOutput, boolInputName);
+    }
+
+    /**
+     * Configures the provided ToolRunner instance with arguments which will invoke the tool represented by this class.
+     * @param toolRunner
+     * @returns {ToolRunner} ToolRunner instance with arguments applied
+     */
+    public configureBuild(toolRunner: ToolRunner): ToolRunner {
+        if (this.isEnabled()) {
+            console.log(tl.loc('codeAnalysis_ToolIsEnabled'), this.toolName);
+            const specifyPluginVersion = tl.getInput('spotbugsGradlePluginVersionChoice') === 'specify';
+            if (specifyPluginVersion) {
+                // #1: Inject custom script to the Gradle build, triggering a Spotbugs run
+                // Add a custom initialisation script to the Gradle run that will apply the Spotbugs plugin and task
+                // Set the Spotbugs Gradle plugin version in the script
+                const pluginVersion: string = this.getSpotBugsGradlePluginVersion();
+                let initScriptPath: string = path.join(__dirname, '..', 'spotbugs.gradle');
+                let scriptContents: string = fs.readFileSync(initScriptPath, 'utf8');
+                scriptContents = scriptContents.replace('SPOTBUGS_GRADLE_PLUGIN_VERSION', pluginVersion);
+                tl.writeFile(initScriptPath, scriptContents);
+                // Specify that the build should run the init script
+                toolRunner.arg(['-I', initScriptPath]);
+            }
+            toolRunner.arg(['check']);
+        }
+        return toolRunner;
+    }
+
+    protected getSpotBugsGradlePluginVersion(): string {
+        let userSpecifiedVersion = tl.getInput('spotbugsGradlePluginVersion');
```

Comment:
use `const` if you don't change variable

Reply:
@matskan  thank you, changed it